src/test/scala/org/broadinstitute/dsde/workbench/sam/api/ResourceRoutesV2Spec.scala

```diff
@@ -311,6 +323,22 @@ class ResourceRoutesV2Spec extends FlatSpec with Matchers with TestSupport with
       accessToCurrentParent = false)
 
     Put(s"/api/resources/v2/${defaultResourceType.name}/${fullyQualifiedChildResource.resourceId.value}/parent", newParentResource) ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.Forbidden
+    }
+  }
+
+  it should "404 if the parent resource does not exist" in {
```

Comment:
This is the new unit test. 

It seems like if a user wanted to find which resource exist or not, they could use setResourceParent, which currently gives a 404 when the resource does not exist and possibly a 403 when it does.

Reply:
I don't think this is right. We want a 403 if the parent resource does not exist. It should behave exactly the same if it exists but there is not access. The user should not be able to distinguish between no access to parent or parent does not exist.

Reply:
looks like ^ is the behavior at the moment, mostly because we pass `None` into `requireParentAction(resource, None` which ends up calling getResourceParent. If we change that, we get the `cyclical resource hierarchy ` error message mentioned in the above comment.